### DISCUSSION TITLE
Import new objects

### DISCUSSION THREAD

**Discussion by Schortenger** [isaac-sim/IsaacLab#48] [Q&A] [answered]
Hi,

Currently I want to build the environment by myself. For example, if I just want to add a cube on the table, can you tell me how to add this basic primitive? 
Moreover, if I have some other complex objects, in URDF or xml format, can you show me how to import them?

**Comment by Mayankm96** [comment, marked as answer]
Hi! 

Sorry for the late response. There are a bunch of tutorials on Isaac Sim / Omniverse side which talks about how to import a new asset. You can find their documentations below:

* URDF Importer: https://docs.omniverse.nvidia.com/app_isaacsim/app_isaacsim/ext_omni_isaac_urdf.html#
* MJCF Importer: https://docs.omniverse.nvidia.com/app_isaacsim/app_isaacsim/ext_omni_isaac_mjcf.html
* ShapeNet Dataset Importer: https://docs.omniverse.nvidia.com/app_isaacsim/app_isaacsim/ext_omni_isaac_shapenet.html
* Asset Converter: https://docs.omniverse.nvidia.com/prod_extensions/prod_extensions/ext_asset-converter.html

An important part to make an asset usable for large-scale parallelization in environments is to make them instanceable. This is a USD concept which helps make asset referencing memory-efficient. You can find more details about this:

* USD SceneGraph Instancing: https://openusd.org/dev/api/_usd__page__scenegraph_instancing.html
* Isaac Sim Instanceable Assets: https://docs.omniverse.nvidia.com/app_isaacsim/app_isaacsim/tutorial_gym_instanceable_assets.html

We are currently working on adding utility scripts that make use of these extensions to make it easier to convert assets using the command line. There has been some progress in this direction:
* `convert_urdf.py`: https://github.com/NVIDIA-Omniverse/Orbit/blob/main/source/tools/convert_urdf.py

We are actively looking for community developers to help us add extensions that readily support converting and importing public object datasets. Hopefully, in the coming period, there will be more progress in this direction. If you're interested to work on this or have some particular datasets you'd like to have available off-the-shelf, feel free to open an issue :)
